src/Page/Auth.elm

```diff
@@ -0,0 +1,698 @@
+module Page.Auth exposing
+    ( Model
+    , Msg(..)
+    , init
+    , update
+    , view
+    )
+
+import Data.Env as Env
+import Data.Session as Session exposing (Session)
+import Dict exposing (Dict)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Http
+import Json.Decode as Decode exposing (Decoder)
+import Json.Decode.Pipeline as Pipe
+import Json.Encode as Encode
+import Ports
+import Views.Alert as Alert
+import Views.Container as Container
+import Views.Markdown as Markdown
+
+
+
+-- Auth flow:
+-- 1/ ask for login:
+--    - ask for a connection link via email: should receive an email with a login link
+--    - once the link in the email received is clicked, the backend will redirect to /#/auth/loggedIn
+--      - GET the user info (to make sure the user is connected)
+--      - load the full processes with impacts
+-- 2/ register:
+--    - ask for registration with email, firstname, lastname, cgu (company): should receive en email with a validation link
+--    - once the link in the email received is clicked, may not go through the login flow
+
+
+login_url : String
+login_url =
+    "/accounts/login.json/"
+
+
+registration_url : String
+registration_url =
+    "/accounts/register.json/"
+
+
+logout_url : String
+logout_url =
+    "/accounts/logout/"
+
+
+profile_url : String
+profile_url =
+    "/accounts/profile.json/"
+
+
+type alias Model =
+    { user : User
+    , response : Maybe Response
+    , action : Action
+    , loggedIn : Bool
+    }
+
+
+type alias User =
+    { email : String
+    , firstname : String
+    , lastname : String
+    , company : String
+    , cgu : Bool
+    , token : String
+    }
+
+
+type alias Form a =
+    { a
+        | next : String
+    }
+
+
+type Response
+    = Success String
+    | Error String (Maybe Errors)
+
+
+type alias Errors =
+    Dict String String
+
+
+formFromUser : User -> Form User
+formFromUser user =
+    { email = user.email
+    , firstname = user.firstname
+    , lastname = user.lastname
+    , company = user.company
+    , cgu = user.cgu
+    , token = ""
+    , next = "/#/auth/loggedIn"
+    }
+
+
+emptyModel : { loggedIn : Bool } -> Model
+emptyModel { loggedIn } =
+    { user =
+        { email = ""
+        , firstname = ""
+        , lastname = ""
+        , company = ""
+        , cgu = False
+        , token = ""
+        }
+    , response = Nothing
+    , action = Register
+    , loggedIn = loggedIn
+    }
+
+
+type Action
+    = Register
+    | Authenticate
+
+
+type Msg
+    = AskForLogin
+    | AskForRegistration
+    | ChangeAction Action
+    | GotUserInfo (Result Http.Error User)
+    | Logout
+    | LoggedIn (Result String Session.AllProcessesJson)
+    | LoggedOut
+    | TokenEmailSent (Result Http.Error Response)
+    | UpdateForm Model
+
+
+init : Session -> { loggedIn : Bool } -> ( Model, Session, Cmd Msg )
```

Comment:
I'd really prefer `authenticated` but I start to sound really annoying with that so from now on I'm gonna skip mentioning it